contracts/0.8.25/vaults/LazyOracle.sol

```diff
@@ -349,10 +361,10 @@ contract LazyOracle is ILazyOracle, AccessControlEnumerableUpgradeable {
         uint256 maxSaneTotalValue = onchainTotalValueOnRefSlot *
             (TOTAL_BASIS_POINTS + $.maxRewardRatioBP) / TOTAL_BASIS_POINTS;
 
+        Quarantine storage q = $.vaultQuarantines[_vault];
+        uint128 quarDelta = q.pendingTotalValueIncrease;
+        uint64 reportTs = $.vaultsDataTimestamp;
         if (_reportedTotalValue > maxSaneTotalValue) {
-            Quarantine storage q = $.vaultQuarantines[_vault];
-            uint64 reportTs = $.vaultsDataTimestamp;
-            uint128 quarDelta = q.pendingTotalValueIncrease;
             uint128 delta = uint128(_reportedTotalValue - onchainTotalValueOnRefSlot);
 
             if (quarDelta == 0) { // first overlimit report
```

Comment:
is it a FSM? probably we can have an ascii art describing states and their transitions